src/sentry/seer/anomaly_detection/get_historical_anomalies.py

```diff
@@ -1,24 +1,164 @@
+    subscription = alert_rule.snuba_query.subscriptions.first()
+    # same deal as above
+    if not subscription:
+        raise ValidationError("No subscription associated with alert rule.")
+    snuba_query = SnubaQuery.objects.get(id=alert_rule.snuba_query_id)
+    dataset = get_dataset(snuba_query.dataset)
+    window_min = int(snuba_query.time_window / 60)
+    start = datetime.fromisoformat(start_string)
+    end = datetime.fromisoformat(end_string)
+    historical_data = fetch_historical_data(
+        alert_rule=alert_rule, snuba_query=snuba_query, project=project, start=start, end=end
+    )
 
-    return [
-        {
-            "timestamp": 0.1,
-            "value": 100.0,
-            "anomaly": {
-                "anomaly_type": AnomalyType.HIGH_CONFIDENCE.value,
-                "anomaly_value": 100,
+    if not historical_data:
+        raise ValidationError("No historical data available.")
+    formatted_data = format_historical_data(historical_data, dataset)
+    if (
+        not alert_rule.sensitivity
+        or not alert_rule.seasonality
+        or alert_rule.threshold_type is None
+        or alert_rule.organization is None
+    ):
+        # this won't happen because we've already gone through the serializer, but mypy insists
+        raise ValidationError("Missing expected configuration for a dynamic alert.")
+
+    anomaly_detection_config = AnomalyDetectionConfig(
+        time_period=window_min,
+        sensitivity=alert_rule.sensitivity,
+        direction=translate_direction(alert_rule.threshold_type),
+        expected_seasonality=alert_rule.seasonality,
+    )
+    try:
+        response = make_signed_seer_api_request(
+            seer_anomaly_detection_connection_pool,
+            SEER_ANOMALY_DETECTION_ENDPOINT_URL,
+            json.dumps(
+                {
+                    "organization_id": alert_rule.organization.id,
+                    "project_id": project.id,
+                    "config": anomaly_detection_config,
+                    "context": formatted_data,
+                }
+            ).encode("utf-8"),
+        )
+    except (TimeoutError, MaxRetryError):
+        logger.warning(
+            "Timeout error when hitting anomaly detection endpoint",
+            extra={
+                "subscription_id": subscription.id,
+                "dataset": alert_rule.snuba_query.dataset,
+                "organization_id": alert_rule.organization.id,
+                "project_id": project.id,
+                "alert_rule_id": alert_rule.id,
             },
-        }
-    ]
+        )
+        return None
+
+    if response.status != 200:
+        logger.error(
+            f"Received {response.status} when calling Seer endpoint {SEER_ANOMALY_DETECTION_ENDPOINT_URL}.",  # noqa
+            extra={"response_data": response.data},
+        )
+        return None
+
+    try:
+        results = json.loads(response.data.decode("utf-8")).get("anomalies")
```

Comment:
The [response type](https://github.com/getsentry/seer/blob/main/src/seer/anomaly_detection/models/external.py#L74-L75) says this is a list of `TimeSeriesPoints` which has singular `anomaly` so to get them all you'd want to iterate through the list and pop off the `anomaly`

Reply:
@nhsiehgit do you need the anomaly itself? Or is sending the timeseriespoints acceptable? What are you expecting in the response?